wpengine/atlas-content-modeler
# Refactor: move field ID collision detection to server

Moves field ID collision detection logic from the client side to the server side:

- For consistency with model/taxonomy name collision detection (which is handled exclusively on the server).
- To make it easier to run field collision detection when fields are created outside of our field UI.
- To simplify upcoming work to allow 'title' and others as a field id (this PR makes no changes to reserved names; allowing 'title' as a field ID will be done in a separate PR).

## Checklist

[No changelog entry required; not a user-facing change.]

## Testing

Adjusted e2e tests to confirm collision detection still works.

To test manually:

- Try to name a field 'id' or 'title'. After submission you should see an error (with WPGraphQL active or inactive).
